agent/agent_test.go

```diff
@@ -292,6 +292,17 @@ func (s *AgentSuite) TestAgentEndTaskShouldExit() {
 	s.Empty(endDetail.FailingCommand, "should not include end task failing command for successful task")
 }
 
+func (s *AgentSuite) TestAgentExitsSingleTaskDistros() {
+	s.setupRunTask(defaultProjYml)
+	s.mockCommunicator.EndTaskResponse = &apimodels.EndTaskResponse{}
+	s.a.opts.SingleTaskDistro = true
+	ctx, cancel := context.WithTimeout(s.ctx, 5*time.Second)
+	defer cancel()
+
+	// The loop should exit after one execution because it is on a single host distro
+	s.NoError(s.a.loop(ctx))
```

Comment:
Can we test if it only looped once somehow?

Reply:
I don't think so 

there isnt a super good test for the agents working in general 😂 because most of them checks that we break under certain conditions

Reply:
Yeah I thought so. I had some ideas of having it run a command and the command modifies a test variable but we don't have anything like that setup and it's non-trivial to do. I'm honestly surprised we haven't had a test command like this yet. This tests looks good to me then!